core/src/main/scala/org/apache/spark/storage/StorageStatusListener.scala

```diff
@@ -74,8 +74,11 @@ class StorageStatusListener(conf: SparkConf) extends SparkListener {
     synchronized {
       val blockManagerId = blockManagerAdded.blockManagerId
       val executorId = blockManagerId.executorId
-      val maxMem = blockManagerAdded.maxMem
-      val storageStatus = new StorageStatus(blockManagerId, maxMem)
+      // This two fields are compatible with old event logs, in which there only has max on heap
+      // memory in the event log. So maxOnHeapMem will use maxMem, maxOffHeapMem will set to 0.
+      val maxOnHeapMem = blockManagerAdded.maxOnHeapMem.getOrElse(blockManagerAdded.maxMem)
```

Comment:
ah I see, thanks for explaining that.  But since that went into spark 2.1, then the meaning of that value in old version could be either, right?